site/src/Librecores/ProjectRepoBundle/Validator/Constraints/UserOrgNameValidator.php

```diff
@@ -37,7 +37,8 @@ class UserOrgNameValidator extends ConstraintValidator
      *
      * @var string[]
      */
-    const RESERVED_NAMES = [ 'org', 'orgs', 'project', 'projects', 'search', 'unassigned' ];
+    const RESERVED_NAMES = [ 'org', 'orgs', 'planet', 'project', 'projects',
+                             'search', 'static', 'unassigned', 'user' ];
 
```

Comment:
Yes we can call $router->match() on the route that get created with `/userName` and see if it matches. E.g in validate() of the user/org name validator use the following:

```php
         if ($this->userOrOrgNameExists($value)) {
            $this->context
                ->buildViolation($constraint->messageUniqueName)
                ->setParameter('%string%', $value)
                ->addViolation();
        }
        else if ($this->userOrOrgReserved($value)) {
            $this->context
                ->buildViolation($constraint->messageReservedName)
                ->setParameter('%string%', $value)
                ->addViolation();
        }
```

where `userOrOrgReserved()` is defined as:

```php
    /**
     * Find out if a given top-level URL value is reserved
     * @param $value
     * @return bool
     */

    private function userOrOrgReserved($value)
    {
        return (in_array($value, self::RESERVED_NAMES) ||
                $this->router->match('/' . $value)['_route']);
    }
```

As we stand now the match() part of the above function is already covered by userOrgNameExists(). However I agree that this is more safe in case we add in the future new routes and forget to update the reserved names of the validator.